device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsIotHubConnection.java

```diff
@@ -688,6 +689,11 @@ else if (remoteState instanceof Modified || remoteState instanceof Released || r
                             this.listener.onMessageSent(inProgressMessages.remove(deliveryTag), transportException);
                         }
                     }
+                    else if (this.inProgressSubscriptionMessages.containsKey(deliveryTag))
+                    {
+                        SubscriptionType subscriptionType = this.inProgressSubscriptionMessages.remove(deliveryTag);
+                        log.debug("Successfully re-sent amqp subscription message of type {}", subscriptionType);
+                    }
                     else
```

Comment:
Without the above changes, this logic would fall to the else block and log an error

Reply:
Why does the logs say: "re-sent amqp subscription message"?
I thought what we do here is -> if we receive an ack from service for the subscription message, we remove it from our internal queue, so as to not resend it. Is that not correct?

Reply:
@timtay-microsoft : A question :raised_hand:

Reply:
This log statement gets executed when we receive an ack for an outstanding subscription message that we sent. It does get removed from the queue of outstanding messages as a result. The only odd thing about this subscription message is that it is sent at the session level. That's why this new queue needed to be added for messages like it